File: src/Flutter/Structs/ErrorBoundaryStruct.cs
using System;
using System.Runtime.InteropServices;
using Flutter.Widgets;

namespace Flutter.Structs
{
    /// <summary>
    /// FFI struct for the ErrorBoundary widget.
    /// </summary>
    [StructLayout(LayoutKind.Sequential)]
    internal class ErrorBoundaryStruct : SingleChildRenderObjectWidgetStruct
    {
        /// <summary>
        /// Whether to show errors in the global error overlay.
        /// </summary>
        public byte showInOverlay { get; set; }

        /// <summary>
        /// Whether to report errors to C# for logging.
        /// </summary>
        public byte reportToNative { get; set; }

        // Has flag for nullable property: widgetTypeName
        public byte HaswidgetTypeName { get; set; }

        // String field: widgetTypeName
        private IntPtr _widgetTypeName;

        /// <summary>
        /// Optional widget type name for error context.
        /// </summary>
        public string? widgetTypeName
        {
            get => GetString(_widgetTypeName);
            set { SetString(ref _widgetTypeName, value); HaswidgetTypeName = (byte)(value != null ? 1 : 0); }
        }

        // Has flag for nullable callback: onError
        public byte HasonError { get; set; }

        // Callback field: onError
        // Using action string pattern - Dart will dispatch action to C# via method channel
        IntPtr _onError;

        /// <summary>
        /// Action identifier for onError callback.
        /// Called when an error is caught in the child widget subtree.
        /// Set to a string identifier (e.g., "action_123") that C# will recognize.
        /// </summary>
        public string? onErrorAction
        {
            get => GetString(_onError);
            set { SetString(ref _onError, value); HasonError = (byte)(value != null ? 1 : 0); }
        }
    }
}

File: flutter_module/lib/structs/errorboundary_struct.dart
import 'dart:ffi';
import 'package:ffi/ffi.dart';
import '../flutter_sharp_structs.dart';

/// FFI struct for the ErrorBoundary widget.
final class ErrorBoundaryStruct extends Struct {
  // FlutterObject Struct base fields
  external Pointer handle;
  external Pointer managedHandle;
  external Pointer<Utf8> widgetType;

  // WidgetStruct base field
  external Pointer<Utf8> id;

  /// The child widget wrapped by this error boundary.
  external Pointer<WidgetStruct> child;

  /// Whether to show errors in the global error overlay.
  @Int8()
  external int showInOverlay;

  /// Whether to report errors to C# for logging.
  @Int8()
  external int reportToNative;

  /// Has flag for widgetTypeName.
  @Int8()
  external int hasWidgetTypeName;

  /// Optional widget type name for error context.
  external Pointer<Utf8> widgetTypeName;

  /// Has flag for onError callback.
  @Int8()
  external int hasOnErrorAction;

  /// Callback ID for the onError callback.
  external Pointer<Utf8> onErrorAction;
}
